### PR TITLE
boards:STM32H7 pad to 256 bit - 32 bytes

### PR DESCRIPTION
**Pull request by davids5**
To address issue noted in https://github.com/PX4/PX4-Autopilot/pull/19690 at the bin file level to remove tools dependency. I.E. https://github.com/PX4/PX4-Autopilot/pull/19690 does not solve the problem for QGC etc.

### CODE REVIEW

**Comment by mhkabir** [comment]
Tested on Durandal with stock uploader, seems to work!

**Review by mhkabir — DISMISSED** [review]

File: boards/cuav/nora/nuttx-config/scripts/script.ld [hunk 1]
```diff
@@ -186,7 +186,10 @@ SECTIONS
 		*(.gnu.linkonce.d.*)
 		CONSTRUCTORS
 		_edata = ABSOLUTE(.);
-	} > AXI_SRAM AT > FLASH
+	    . = ALIGN(8);
```
Comment (bkueng) [inline boards/cuav/nora/nuttx-config/scripts/script.ld:189]: I'd add a comment here as to why this is required

**Comment by spiderkeys** [comment]
@davids5 is https://github.com/PX4/PX4-Autopilot/pull/19690 still required when using px_uploader.py, or is this change to the linker script sufficient? 

I've tested with both PR's changes on the Matek H743 Wing, and it seems to have resolved the hardfault issues for me, but it sounds like the change to px_uploader.py may be unnecessary?

File: boards/cuav/nora/nuttx-config/scripts/script.ld (hunk 1 above)
Comment (davids5) [reply boards/cuav/nora/nuttx-config/scripts/script.ld]: @bkueng - Thank you! I mad me realize it is 256 bit - 8 32 bit words

**Comment by davids5** [comment]
> @davids5 is #19690 still required when using px_uploader.py, or is this change to the linker script sufficient?
> 
> I've tested with both PR's changes on the Matek H743 Wing, and it seems to have resolved the hardfault issues for me, but it sounds like the change to px_uploader.py may be unnecessary?

Only the linker script change is need.

**Comment by davids5** [comment]
@mhkabir @spiderkeys @Abner0907 - In adding @bkueng requested comment mad me realize I had the sizing wrong.  Please retest. This change.

**Comment by spiderkeys** [comment]
> @mhkabir @spiderkeys @Abner0907 - In adding @bkueng requested comment made me realize I had the sizing wrong. Please retest. This change.

Just tested with the new changes, and it works as far as I can tell. The issue used to be intermittent, so hard to say 100%.

File: boards/cuav/nora/nuttx-config/scripts/script.ld [hunk 2]
```diff
@@ -186,7 +186,13 @@ SECTIONS
...
 		CONSTRUCTORS
 		_edata = ABSOLUTE(.);
-	} > AXI_SRAM AT > FLASH
+
+	    /* Pad out last section as the STM32H7 Flash write size is 256 bits. 32 bytes */
+
+	    . = ALIGN(32);
+		FILL(0xff)
+		. += 32;
+	} > AXI_SRAM AT > FLASH  = 0xff
```
Comment (bkueng) [inline boards/cuav/nora/nuttx-config/scripts/script.ld:195]: According to this [doc](https://ftp.gnu.org/old-gnu/Manuals/ld-2.9.1/html_chapter/ld_3.html), the fill should be 2 bytes:
> =fill
    Including =fill in a section definition specifies the initial fill value for that section. You may use any expression to specify fill. Any unallocated holes in the current output section when written to the output file will be filled with the two least significant bytes of the value, repeated as necessary. You can also change the fill value with a FILL statement in the contents of a section definition. 

```suggestion
	} > AXI_SRAM AT > FLASH  = 0xffff
```

Comment (davids5) [reply boards/cuav/nora/nuttx-config/scripts/script.ld]: Yes. Good catch. It was making a the 8bit 0xff into a 16 bit of 0xffff automatically 

but it is worse then that.  Padding is 16 bit. So the size needs to be 16 for 16 bit words not bytes.

Comment (davids5) [reply boards/cuav/nora/nuttx-config/scripts/script.ld]: Master:
```
001de930: 0000 0000 0000 0000 0000 0000 5636 5878  ............V6Xx
001de940: 7800 0000 4816 0024 0100 0000 0000 0000  x...H..$........
001de950: 0000 0000 0000 0000 0000 0000 ff         .............
```

This PR

```
01de960: 0000 0000 0000 0000 0000 0000 5636 5878  ............V6Xx
001de970: 7800 ffff 4816 0024 0100 0000 0000 0000  x...H..$........
001de980: 0000 0000 0000 0000 0000 0000 ffff ffff  ................
001de990: ffff ffff ffff ffff ffff ffff ffff ffff  ................
001de9a0: ffff ffff ffff ffff ffff ffff            ............
```

**Review by bkueng — APPROVED** [review]

**Comment by davids5** [comment]
STM32H7 Bootloaders are fixed here https://github.com/PX4/PX4-Autopilot/pull/19728

**Comment by davids5** [comment]
rebased on master

**Review by dagar — APPROVED** [review]

**Pull request merged**